databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,471 @@
+        for node_file in self._node_files:
+            self._create_indices(node_file=node_file)
+
+        LOGGER.info('Publishing Node files: %s', self._node_files)
+        while True:
+            try:
+                node_file = next(self._node_files_iter)
+                self._publish_node(node_file)
+            except StopIteration:
+                break
+
+        LOGGER.info('Publishing Relationship files: %s', self._relation_files)
+        while True:
+            try:
+                relation_file = next(self._relation_files_iter)
+                self._publish_relation(relation_file)
+            except StopIteration:
+                break
+
+        LOGGER.info('Committed total %i statements', self._count)
+
+        # TODO: Add statsd support
+        LOGGER.info('Successfully published. Elapsed: %i seconds', time.time() - start)
+
+    def get_scope(self) -> str:
+        return 'publisher.neo4j'
+
+    def _create_indices(self, node_file: str) -> None:
+        """
+        Go over the node file and try creating unique index
+        :param node_file:
+        :return:
+        """
+        LOGGER.info('Creating indices. (Existing indices will be ignored)')
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            for node_record in pandas.read_csv(node_csv,
+                                               na_filter=False).to_dict(orient='records'):
+                label = node_record[NODE_LABEL]
+                if label not in self._labels:
+                    with self._driver.session(database=self._db_name) as session:
+                        try:
+                            session.write_transaction(self._try_create_index, label)
+                        except Neo4jError as e:
+                            if 'An equivalent constraint already exists' not in e.__str__():
+                                raise
+                            # Else, swallow the exception, to make this function idempotent.
+                    self._labels.add(label)
+
+        LOGGER.info('Indices have been created.')
+
+    def _write_transactions(self,
+                            records: List[dict],
+                            stmt: str) -> None:
+        params_list = []
+        start_idx = 0
+        while start_idx < len(records):
+            stop_idx = min(start_idx + self._transaction_size, len(records))
+
+            for i in range(start_idx, stop_idx):
+                params_list.append(self._create_props_param(records[i]))
+
+            with self._driver.session(database=self._db_name) as session:
+                session.write_transaction(self._execute_statement, stmt, {'batch': params_list})
+
+            params_list.clear()
+            start_idx = start_idx + self._transaction_size
+
+    def _publish_node(self, node_file: str) -> None:
+        """
+        Iterate over the csv records of a file, each csv record transform to Merge statement
```

Comment:
let's remove this lengthy comment. same for _publish_rel. The actual code is a better documentation than this re-statement of it.